worker/diskmanager/lsblk.go

```diff
@@ -218,6 +220,14 @@ func addHardwareInfo(dev *storage.BlockDevice) error {
 		return errors.Annotate(err, "cannot parse udevadm output")
 	}
 
+	// For cases where there are logical disks attached to a
+	// controller (eg RAID), the controller itself has a WWN
+	// which is the ID_WWN value and each disk has a WWN with
+	// a vendor extension ID_WWN_WITH_EXTENSION added to
+	// identify the disk via the /dev/disk/by-id path.
+	if wwnWithExtension != "" {
```

Comment:
Just checking I understand - the reason for doing this here rather than in the ID_WWN_WITH_EXTENSION case is because that and ID_WWN might be in the output in either order, and we always want the extended version to take precedence if it's present. Is that right?

Reply:
edzakery